daffodil-runtime1/src/main/scala/org/apache/daffodil/infoset/XMLTextInfosetOutputter.scala

```diff
@@ -156,8 +158,11 @@ class XMLTextInfosetOutputter private (writer: java.io.Writer, pretty: Boolean)
 
     if (!isNilled(simple) && simple.hasValue) {
       if (simple.erd.optPrimType.get == NodeInfo.String) {
-        val simpleVal = simple.dataValueAsString
-        if (simple.erd.runtimeProperties.get(XMLTextInfoset.stringAsXml) == "true") {
+        var simpleVal = simple.dataValueAsString
+        if (xmlOutputStyle == XMLOutputStyle.PrettyPrintSafe){
+          simpleVal = "<![CDATA[%s]]>".format(remapped(simpleVal).replaceAll("]]>", "]]]]><![CDATA[>"))
+          writer.write(simpleVal)
+        } else if (simple.erd.runtimeProperties.get(XMLTextInfoset.stringAsXml) == "true") {
           writeStringAsXml(simpleVal)
         } else {
           writer.write(scala.xml.Utility.escape(remapped(simpleVal)))
```

Comment:
For easier maintainability, I suggest mutating simpleVal and calling write.write(simpleVal) here too.